src/cavegen.cpp

```diff
@@ -32,16 +31,16 @@ NoiseParams nparams_caveliquids(0, 1, v3f(150.0, 150.0, 150.0), 776, 3, 0.6, 2.0
 ///////////////////////////////////////// Caves V5
 
 
-CaveV5::CaveV5(MapgenV5 *mg, PseudoRandom *ps)
+CaveV5::CaveV5(Mapgen *mg, PseudoRandom *ps)
 {
-	this->mg             = mg;
+	this->mg             = (MapgenV5 *)mg;
```

Comment:
This may work by chance, but if you change either of these classes (MapgenV5 or MapgenFractal) then it won't anymore. You should either use a dynamic_cast for both, and check which is not null, or implement two different functions for this.